recipe/gen_patch_json.py

```diff
@@ -808,8 +808,25 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
         if i >= 0:
             deps[i] = "cudatoolkit >=11.2,<12.0a0"
 
-        if "libiconv >=1.15,<1.16.0a0" in deps:
-            _pin_looser(fn, record, "libiconv", upper_bound="1.17.0")
+        if record.get('timestamp', 0) < 1662860143000:
+            if any(dep.startswith("arpack >=3.7") for dep in deps):
+                _pin_looser(fn, record, "arpack", max_pin="x.x")
+            if any(dep.startswith("libiconv >=1") for dep in deps):
+                _pin_looser(fn, record, "libiconv", max_pin="x")
+            if any(dep.startswith("cairo >=1") for dep in deps):
+                _pin_looser(fn, record, "cairo", max_pin="x")
+            if any(dep.startswith("glpk >=5") for dep in deps):
+                _pin_looser(fn, record, "glpk", max_pin="x")
+            if any(dep.startswith("nlopt >=2.7") for dep in deps):
+                _pin_looser(fn, record, "nlopt", max_pin="x.x")
+            if any(dep.startswith("openjpeg >=2.4") for dep in deps):
+                _pin_looser(fn, record, "openjpeg", max_pin="x")
+            if any(dep.startswith("pango >=1.48") for dep in deps):
+                _pin_looser(fn, record, "pango", max_pin="x")
+            if any(dep.startswith("pango >=5.2") for dep in deps):
+                _pin_looser(fn, record, "xz", max_pin="x")
+            if any(dep.startswith("libxml2 >=2.9") for dep in deps):
+                _pin_looser(fn, record, "libxml2", upper_bound="2.11.0")
```

Comment:
Are we going to have to set the upper bound up every time?

Reply:
Yes, unless you feel confident to change the run_exports to `x` in libxml2.

Reply:
your other comment addressed my question.